File: 0137-single-number-ii/0137-single-number-ii.java
class Solution {
    public int singleNumber(int[] nums) {
        Map<Integer, List<Integer>> map = Arrays.stream(nums)
                .mapToObj(i -> i)
                .collect(Collectors.groupingBy(i -> i));
        
        Iterator<Integer> iter = map.keySet().iterator();
        while(iter.hasNext()) {
            int key = iter.next();
            List<Integer> list = map.get(key);
            if (list.size() == 1) {
                return key;
            }
        }
        return 0;
    }
}

File: 0137-single-number-ii/0137-single-number-ii.go
func singleNumber(nums []int) int {
    m := make(map[int]int)
	for _, k := range nums {
		_, ok := m[k]
		if ok {
			m[k] = m[k] + 1
		} else {
			m[k] = 1
		}
	}

	for key := range m {
		if m[key] == 1 {
			return key
		}
	}

	return 0
}